content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+Try using the visualisation diagram above to work out what happens when the computer executes the return statement.
+
+{{</note>}}
+
+### 🧩 Putting everything together
+
+So now we have some idea of how the computer executes our code, we can apply these ideas to a program for our current implementation of `convertToPercentage`:
+
+<iframe width="800" height="350" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=function%20convertToPercentage%28%29%20%7B%0A%20%20const%20num%20%3D%200.5%3B%0A%20%20const%20percentage%20%3D%20%60%24%7Bnum%20*%20100%7D%25%60%3B%0A%20%20return%20percentage%3B%0A%7D%0A%0Aconst%20output1%20%3D%20convertToPercentage%280.5%29%3B%0Aconst%20output2%20%3D%20convertToPercentage%280.3%29%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=0&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+{{<note type="activity" title="exercise">}}
+
+Earlier we noted that when the computer executes the code above, we get the following output:
+
+```bash
+50%
+50%
+```
+
+Given the code above, try using the visualisation diagrams to "play computer" with this example. Try stepping through each line of code to explain why the program is giving this output
+
+{{</note>}}
+
+### Checking our code
+
+Given our code below:
+
+```js {linenos=table,hl_lines=[7] ,linenostart=1}
+function convertToPercentage() {
+  const num = 0.5;
+  const percentage = `${num * 100}%`;
+  return percentage;
+}
+
+const output1 = convertToPercentage(0.5);
+const output2 = convertToPercentage(0.23);
+```
+
+when the function `convertToPercentage` is called, the variable `num` gets recreated in the `convertToPercentage` frame and set to `0.5` every time we call the function. Instead of `num` being reset every time the function is called, we want to make use of the input that is passed to the function `convertToPercentage` every time it is called.
+
+### Parametrising a function
+
+To make this function reusable for any number, we need to handle inputs. We do this using a **parameter**.
+
+🔑 A parameter is a variable that enables us to reference the input to a function.
+
+We can add a parameter `num` to our function, as with the execution diagram below:
+
+We can think of a function as a "box". We pass inputs to into it, the function's code is executed and we get an output at the end. We can visualise this as follows:
+
+```mermaid
+
+flowchart LR
+    A[input] --> B{function}
+    B --> C[output]
+```
+
+We could visualise what happens when `makeGreeting` is passed a specific argument:
+
+```mermaid
+
+flowchart LR
+    A[0.23] --> B{convertToPercentage}
+    B --> C[23%]
+```
+
+This time we have a difference in that we have defined a parameter `num` in the function declaration inside parentheses after the function name `convertToPercentage`. In our current mental model, a function call means going to `convertToPercentage` and running the code inside the function.
```

Comment:
What about the fact that functions can have more than one argument? Once we've established the basic mental model for arguments, it seems to me like the cognitive load of showing that arguments can be multiple at this point is minimal. We could revisit the examples of area and perimeter to show how to use multiple arguments.

Reply:
I agree the cognitive load is minimal - and that's why I've not left it in here. There will be exercises where they'll encounter this concept and if their mental model is sufficiently good enough then they'll be able to pick up the idea of 2 parameters without us explaining it here.
I'm aiming to build a minimal viable mental model here.